.github/workflows/infrastructure-tests.yml

```diff
@@ -0,0 +1,118 @@
+
+      - name: Set up dotnet
+        uses: actions/setup-dotnet@607fce577a46308457984d59e4954e075820f10a # v3.0.3
+        with:
+          dotnet-version: '6.0.x'
+
+      - name: Restore Tools
+        run: dotnet tool restore
+
+      - name: Compose Databases
+        working-directory: 'dev'
+        # We could think about not using profiles and pulling images directly to cover multiple versions
+        run: |
+          cp .env.example .env
+          docker compose --profile mssql --profile postgres --profile mysql up -d
+        shell: pwsh
+      
+      # I've seen the SQL Server container not be ready for commands right after starting up and just needing a bit longer to be ready
+      - name: Sleep
+        run: sleep 15s
+
+      - name: Migrate SQL Server
+        working-directory: 'dev'
+        run: "pwsh ./migrate.ps1"
+        shell: pwsh
+
+      - name: Migrate MySQL
+        working-directory: 'util/MySqlMigrations'
+        run: 'dotnet ef database update --connection "$CONN_STR" -- --GlobalSettings:MySql:ConnectionString="$CONN_STR"'
+        env:
+          CONN_STR: "server=localhost;uid=root;pwd=SET_A_PASSWORD_HERE_123;database=vault_dev;Allow User Variables=true"
+
+      - name: Migrate Postgres
+        working-directory: 'util/PostgresMigrations'
+        run: 'dotnet ef database update --connection "$CONN_STR" -- --GlobalSettings:PostgreSql:ConnectionString="$CONN_STR"'
+        env:
+          CONN_STR: "Host=localhost;Username=postgres;Password=SET_A_PASSWORD_HERE_123;Database=vault_dev"
+      
+      - name: Migrate Sqlite
+        working-directory: 'util/SqliteMigrations'
+        run: 'dotnet ef database update --connection "$CONN_STR" -- --GlobalSettings:Sqlite:ConnectionString="$CONN_STR"'
+        env:
+          CONN_STR: "Data Source=${{ runner.temp }}/test.db"
+        
+      - name: Run Tests
+        working-directory: 'test/Infrastructure.IntegrationTest'
+        env:
+          # Default Postgres:
+          BW_TEST_DATABASES__0__TYPE: "Postgres"
+          BW_TEST_DATABASES__0__CONNECTIONSTRING: "Host=localhost;Username=postgres;Password=SET_A_PASSWORD_HERE_123;Database=vault_dev"
+          # Default MySql
+          BW_TEST_DATABASES__1__TYPE: "MySql"
+          BW_TEST_DATABASES__1__CONNECTIONSTRING: "server=localhost;uid=root;pwd=SET_A_PASSWORD_HERE_123;database=vault_dev"
+          # Default Dapper SqlServer
+          BW_TEST_DATABASES__2__TYPE: "SqlServer"
+          BW_TEST_DATABASES__2__CONNECTIONSTRING: "Server=localhost;Database=vault_dev;User Id=SA;Password=SET_A_PASSWORD_HERE_123;Encrypt=True;TrustServerCertificate=True;"
+          BW_TEST_DATABASES__2__USEEF: "false" # Defaults to false so this isn't needed but this is how it is possible to also run the EF version of SqlServer by setting this to false
+          # Default Sqlite
+          BW_TEST_DATABASES__3__TYPE: "Sqlite"
+          BW_TEST_DATABASES__3__CONNECTIONSTRING: "Data Source=${{ runner.temp }}/test.db"
```

Comment:
I'd like to see dapper configured in the same way as EF databases here. I don't see why it would be a special case that is implied